src/pages/auth/hooks/useEmailPrompt.ts

```diff
@@ -0,0 +1,77 @@
+import { useMutation } from '@apollo/client'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { useAuthContext } from '../../../context'
+import { MUTATION_UPDATE_USER_EMAIL } from '../../../graphql/mutations/email'
+
+type FormValues = Record<string, any>
+
+const emailSchema = yup.object().shape({
+  dontAskAgain: yup.boolean(),
+  email: yup
+    .string()
+    .email('Invalid email')
+    .when('dontAskAgain', {
+      is: false,
+      then: (schema) => schema.required('Email is required'),
+      otherwise: (schema) => schema.notRequired(),
+    }),
+})
+
+export const useEmailPrompt = () => {
+  const { user, setUser, isLoggedIn } = useAuthContext()
+  const [shouldPrompt, setShouldPrompt] = useState(false)
+  const [enableSave, setEnableSave] = useState(false)
+  const [updateUserEmail] = useMutation(MUTATION_UPDATE_USER_EMAIL)
```

Comment:
let's eliminate the use of individual query/mutation use cases, and use available hooks for them, that way they are typed.
```suggestion
  const [updateUserEmail] = 
  useUpdataUserEmailMutation()
```